index.js

```diff
@@ -197,20 +199,30 @@ async function underPressure (fastify, opts) {
   }
 
   async function onStatus (req, reply) {
+    let response = { status: 'ok' }
     if (healthCheck) {
       try {
-        if (!await healthCheck()) {
+        const checkResult = await healthCheck()
+        if (!checkResult) {
           req.log.error('external health check failed')
           reply.status(SERVICE_UNAVAILABLE).header('Retry-After', retryAfter)
           throw underPressureError
         }
+
+        if (typeof checkResult === 'object' && checkResult !== null && !Array.isArray(checkResult)) {
+          response = {
+            ...response,
+            ...checkResult
+          }
+        }
       } catch (err) {
         req.log.error({ err }, 'external health check failed with error')
         reply.status(SERVICE_UNAVAILABLE).header('Retry-After', retryAfter)
         throw underPressureError
       }
     }
-    return { status: 'ok' }
+
+    return response
```

Comment:
let's simplify this code.

1. no need to define a variable with let at the top when you're just returning it
2. no need for all those checks at line 212. when you reach that point, you know that checkResult has a truthy value. let's just spread it onto the response.